src/main/java/everyos.browser.webicitybrowser/everyos/browser/webicitybrowser/gui/ui/OverlyingBlockComponentUI.java

```diff
@@ -1,72 +1,103 @@
         super(c, parent);
-    }
-
-    //TODO: Fix merge conflicts
-    
-    /*@Override
-    protected ComponentUI[] calcChildren(UIManager uimgr) {
-        component = getComponent();
-        if (component.getChildren().length < 1) return new ComponentUI[0];
-        contentPane = component.getChildren()[0];
-        contentPane.unbindAll();
-        contentPaneUI = uimgr.get(contentPane, getParent());
-        contentPane.bind(contentPaneUI);
-        return new ComponentUI[]{contentPaneUI};
+        component = (OverlyingBlockComponent) c;
+        this.appearence = new OverlyingBlockComponentUIAppearence();
+        layout = (InlineBlockLayout) getLayout();
+        layout.setConsiderChildren(false);
     }
 
     @Override
-    protected ComponentUI[] getChildren() {
-        return new ComponentUI[]{contentPaneUI};
+    public Appearence getAppearence() {
+        return appearence;
     }
 
-    @Override
-    protected void paintUI(Renderer r) {
-        paintMouse(r);
+    private class OverlyingBlockComponentUIAppearence implements Appearence {
+
+        private Dimension bounds;
+
+        @Override
+        public void render(Renderer r, SizePosGroup sizepos, UIManager uimgr) {
+//            sizepos.move(strwidth+r.getFontPaddingHeight(), true);
+//            sizepos.setMinLineHeight(r.getFontHeight());
+
+            this.bounds = sizepos.getSize();
+            layout.renderChildren(r, sizepos, uimgr);
+
+//            contentPaneUI = uimgr.get(component.contentView, OverlyingBlockComponentUI.this);
+//            contentPaneUI.render(r, sizepos, uimgr);
+//            contentPaneUI.directive(SizeDirective.of(new Location(1, 0, 1, 0)).getDirective());
+        }
 
-        if(component.isInvisible()) return;
+        private float progress = 0;
 
-        Rectangle bounds = getBounds();
+        @Override
+        public void paint(Renderer r) {
+            if (component.isInvisible()) {
+                if (progress > 0) progress -= TimeSystem.getDeltaSeconds();
+                else return;
+            } else {
+                if (progress < 1)
+                    progress += TimeSystem.getDeltaSeconds();
+            }
 
-        r.useBackground();
-        int w = bounds.getWidth();
-        int h = bounds.getHeight();
-        int d = Styling.BUTTON_WIDTH;
-        int l2 = d / 2;
-        int ws = w - d;
-        int hs = h - d;
-        r.drawEllipse(0, 0, d, d);
-        r.drawEllipse(ws, 0, d, d);
-        r.drawEllipse(ws, hs, d, d);
-        r.drawEllipse(0, hs, d, d);
-        r.drawFilledRect(l2, 0, ws, d);
-        r.drawFilledRect(ws, l2, d, hs);
-        r.drawFilledRect(l2, hs, ws, d);
-        r.drawFilledRect(0, l2, d, hs);
-        r.drawFilledRect(d, d, w - 2 * d, h - 2 * d);
+            r.useBackground();
+            int w = bounds.getWidth();
+            int h = bounds.getHeight();
 
-        r.useForeground();
+            int ha = (int) (h * progress);
```

Comment:
What does ha stand for? Any chance the variable can be renamed?

Reply:
😅
Okay... I can rename it to one of
heightAnimated
hAnimated 
h_anim
h_eff
hEffective
heightEffective
And so on.

Reply:
heightAnimated sounds good